src/master.rs

```diff
@@ -148,11 +150,25 @@ impl Master {
     pub fn get_info(&self) -> Result<MasterInfo> {
         let mut data = ec::ec_ioctl_master_t::default();
         ioctl!(self, ec::ioctl::MASTER, &mut data)?;
+        let ec::ec_ioctl_master_t {
+            slave_count,
+            devices,
+            scan_busy,
+            app_time,
+            ..
+        } = data;
+        let link_up = devices
+            .iter()
+            .nth(0)
+            .ok_or_else(|| Error::new(ErrorKind::InvalidData, "No devices available"))?
+            .link_state
+            != 0;
```

Comment:
This rustfmt-ism is not very readable to me.  What about `let first_dev = ...?; let link_up = first_dev.link_state != 0`?


Also, `iter().nth(0)` -> `get(0)`?